exonum-java-binding-core/src/test/java/com/exonum/binding/storage/indices/AbstractIndexProxyTest.java

```diff
@@ -74,23 +57,31 @@ void constructorFailsIfNullView() {
   @Test
   void notifyModifiedThrowsIfSnapshotPassed() {
     Snapshot dbView = createSnapshot();
-    proxy = new IndexProxyImpl(dbView);
+    ModificationCounter counter = dbView.getModificationCounter();
+    int initialValue = counter.getCurrentValue();
 
-    Pattern pattern = Pattern.compile("Cannot modify the view: .*[Ss]napshot.*"
-        + "\\nUse a Fork to modify any collection\\.", Pattern.MULTILINE);
+    proxy = new IndexProxyImpl(dbView);
 
     UnsupportedOperationException thrown = assertThrows(UnsupportedOperationException.class,
         () -> proxy.notifyModified());
+
+    Pattern pattern = Pattern.compile("Cannot modify the view: .*[Ss]napshot.*"
+        + "\\nUse a Fork to modify any collection\\.", Pattern.MULTILINE);
     assertThat(thrown.getMessage(), matchesPattern(pattern));
+
+    assertFalse(counter.isModifiedSince(initialValue));
   }
 
   @Test
   void notifyModifiedAcceptsFork() {
```

Comment:
This test verifies that the index does use a counter from the Fork. Not sure if we shall add an integration test of Collection + Iterator + ModificationCounter, verifying that modification of another collection created with the same Fork invalidates the iterator of the first collection.